include/sample_rdd.h

```diff
@@ -0,0 +1,92 @@
+#ifndef CPARK_SAMPLE_RDD_H
+#define CPARK_SAMPLE_RDD_H
+
+#include <random>
+#include <vector>
+#include "base_rdd.h"
+#include "filter_rdd.h"
+#include "utils.h"
+namespace cpark {
+
+/**
+ * An Rdd holding the sampled data from an old rdd by a specific sample fraction
+ * @tparam R Type of the old Rdd.
+ * The type of the old Rdd `R`'s elements should be able to invoke random boolean function,
+ * the invoke result must be in boolean, and the element type shouldn't change after sampling.
+ */
+template <concepts::Rdd R>
+requires std::invocable<std::function<bool(int)>, utils::RddElementType<R>>&& std::is_convertible_v<
+    std::invoke_result_t<std::function<bool(int)>, utils::RddElementType<R>>, bool> class SampleRdd
+    : public BaseRdd<SampleRdd<R>> {
+public:
+  using Base = BaseRdd<SampleRdd<R>>;
+  friend Base;
+
+public:
+  /**
+   * Main constructor of SampleRdd.
+   * @param prev Reference to previous Rdd of type R
+   * @param fraction The desired sampling ratio from the 'prev' RDD
```

Comment:
How about also documenting what will happen if `fraction` does not fall in range `[0,1]`?